pkg/database/gorm.go

```diff
@@ -1,18 +1,54 @@
 package database
 
 import (
+	"fmt"
 	"strconv"
 	"time"
 
 	"github.com/DATA-DOG/go-txdb"
+	gormtrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/gorm.io/gorm.v1"
 	"gorm.io/driver/mysql"
 	"gorm.io/gorm"
 )
 
 const testEnv = "test"
 
-// NewConnection returns a new Gorm database connection.
+// NewInstrumentedConnection returns a new instrumented Gorm database connection.
+func NewInstrumentedConnection(config *Config, environment, appName string) (*gorm.DB, error) {
+	return newConnection(config, environment, appName, true)
+}
+
+// NewConnection returns a new uninstrumented Gorm database connection.
 func NewConnection(config *Config, environment string) (*gorm.DB, error) {
+	return newConnection(config, environment, "", false)
+}
```

Comment:
I believe we should always return an instrumented connection. Previously, as far as I understand, `NewConnection` was instrumented. Now it's not. Was the idea behind introducing `NewInstrumentedConnection` to avoid breaking the `NewConnection` interface? 🤔

Reply:
`NewConnection` was uninstrumented, instead we were using [InstrumentDatabase](https://github.com/scribd/go-sdk/pull/106/files#diff-2b52422b8ec8cbd7b7316c2553325d0c7e8bea0b005139182447457e8c365ec3L36) called in [main.go](https://github.com/scribd/go-chassis/blob/main/cmd/server/main.go#L95).
We decided to add another function to follow the harmony. Usually what happens is that we have a secondary function to instrument a default client (ex [AWSClient](https://github.com/scribd/go-chassis/blob/main/cmd/server/main.go#L108)), but since the `dd-trace-go` library wasn't flexible enough, we decided to separate the funcs.

But, I agree every client should be instrumented, and maybe we could just add a flag in config to switch it off/on. Gonna be a big work as we have an [instrumentation](https://github.com/scribd/go-sdk/tree/main/pkg/instrumentation) package with many of funcs in there to be repurposed.

Reply:
Based on the discussion [here](https://github.com/scribd/go-sdk/pull/106#discussion_r1637852856), now there is only one `NewConnection` and it's instrumented.